File: portal/portal/App_Code/DAO/GrupoAcessoDAO.cs
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using MySql.Data.MySqlClient;
using System.IO;

namespace portal.App_Code.DAO
{
    public class GrupoAcessoDAO
    {
        //lendo string de conexão     
        String sc = Properties.Settings.Default.CN;
        String logPath = Properties.Settings.Default.PathErrorLog;

        public GrupoAcesso carregar(long pCodigo)
        {
            GrupoAcesso obj = new GrupoAcesso();
            try
            {
                //cria a conexao com o bd
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from grupo_acesso where codigo=" + pCodigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                if (registros.Read())
                {
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Nome = registros["nome"].ToString();
                    obj.Status = (GrupoAcesso.TipoStatus)registros["status"];
                    obj.Paginas = PaginaAcessoDAO.carregarLista(obj.Codigo);
                }
                conexao.Close();
            }
            catch (Exception err)
            {
                String log = "Erro=>" + DateTime.Now + err.Message + Environment.NewLine;
                File.AppendAllText(logPath, log);
            }
            return obj;
        }

        public List<GrupoAcesso> carregarLista(String[] pFiltros, String pOrdena)
        {
            List<GrupoAcesso> lista = new List<GrupoAcesso>();
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();
                String sql = "select * from grupo_acesso where ";
                for (int i = 0; i < pFiltros.Length; i++)
                {
                    sql = sql + pFiltros[i];
                }
                sql = sql + " order by " + pOrdena;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                MySqlDataReader registros = comando.ExecuteReader();
                while (registros.Read())
                {
                    GrupoAcesso obj = new GrupoAcesso();
          
                    obj.Codigo = Convert.ToInt64(registros["codigo"]);
                    obj.Nome = registros["nome"].ToString();
                    obj.Status = (GrupoAcesso.TipoStatus)registros["status"];

                    obj.Paginas = PaginaAcessoDAO.carregarLista(obj.Codigo);
                    lista.Add(obj);
                }
                conexao.Close();
            }
            catch (Exception err)
            {

            }
            return lista;
        }

        public void persistir(GrupoAcesso obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "";
                if (obj.Codigo == 0)
                {
                    sql = "insert into grupo_acesso(nome, status) value('{0}',{1})";
                    sql = String.Format(sql, obj.Nome,  obj.Status);
                }
                else
                {
                    sql = "update grupo_acesso set nome='{0}', status='{1}' where codigo={2}";
                    sql = String.Format(sql, obj.Nome, obj.Status, obj.Codigo);
                }
                //gravar lista de paginas
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }


        public void remover(GrupoAcesso obj)
        {
            try
            {
                MySqlConnection conexao = new MySqlConnection(sc);
                conexao.Open();

                string sql = "delete from grupo_acesso where codigo=" + obj.Codigo;
                MySqlCommand comando = new MySqlCommand(sql, conexao);
                comando.ExecuteNonQuery();
                conexao.Close();
            }
            catch (Exception err)
            {

            }
        }

    }
}

File: portal/portal/App_Code/DAO/GrupoAcessoDAO.php
<?php
namespace portal\App_Code\DAO {
    
//CHAMADAS EXTERNAS------------------------------------------------------------------------------------------------------------------------------
use portal\App_Code\GrupoAcesso;
use portal\App_Code\PaginaAcesso;
use portal\App_Code\DAO\PaginaAcessoDAO;
mysqli_report(MYSQLI_REPORT_STRICT|MYSQLI_REPORT_ERROR);

    try{
		require_once("/../GrupoAcesso.php");  
        require_once("/../PaginaAcesso.php");        
        require_once("/../../connection.php");        
    }catch(Exception $e){
		require_once (dirname(__file__, 2) . "\GrupoAcesso.php");
		require_once (dirname(__file__, 2) . "\PaginaAcesso.php");
		require_once (dirname(__file__, 3) . "\connection.php");
    }

class GrupoAcessoDAO{
//INSERIR----------------------------------------------------------------------------------------------------------------------------------------
    function inserirGrupoAcesso($groupAcess){
        try{
            $con = BD_AbrirConexao();
            $sql = "insert into grupo_acesso (nome,status) values (' " .
				 $groupAcess->getNome() . " ', " .
                 $groupAcess->getTipoStatus() . ")";
                 mysqli_query($con, $sql);
                 echo "Inserção com sucesso!";
        }catch (Exception $e){
                echo "Erro de inserção: " . $e;
        }
    mysqli_close($con);	
		}

//ALTERAR---------------------------------------------------------------------------------------------------------------------------------------
    function alterarGrupoAcesso($groupAcess){  
        try{
            $con = BD_AbrirConexao();
            $sql = "update grupo_acesso set " . 
                "nome = ' " . $groupAcess->getNome() . "', " .
                "status = " . $groupAcess->getTipoStatus() .
                " where codigo = " . $groupAcess->getCodigo();
                mysqli_query($con,$sql);
				echo "Dados alterados com sucesso";
        }catch (Exception $e){
                echo "Erro ao alterar dados: " . $e;
                }
    mysqli_close($con);	
    }

//CARREGAR--------------------------------------------------------------------------------------------------------------------------------------
function carregarGrupoAcesso($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "select * from grupo_acesso where codigo = " . $codigo;
                $gargantula = mysqli_query($con,$sql);
                $gpAcess = new GrupoAcesso();				
                    if($carga = mysqli_fetch_array($gargantula)){
                        $gpAcess->setGrupoAcesso(($carga["codigo"]),
												 ($carga["nome"]),
												 ($carga["status"]));
						mysqli_close($con);
						return $gpAcess;
                    }else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;             
            }
       mysqli_close($con);
    }

    function carregarGrupoAcessoComPaginas($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "select * from grupo_acesso where codigo = " . $codigo;
                $gargantula = mysqli_query($con,$sql);
                    if($carga = mysqli_fetch_array($gargantula)){
					   $gpAcess = new GrupoAcesso();
				       $daoPagina = new PaginaAcessoDAO();
                       $gpAcess->setGrupoAcessoComPaginas(($carga["codigo"]),
												          ($carga["nome"]),
														  ($carga["status"]),
														  ($daoPagina->carregarListaPaginaAcessoCod($codigo)));
						mysqli_close($con);
						return $gpAcess;
                    }else{
                        echo "Registros não encontrados!";
                    }
            }catch (Exception $e){
                echo "Erro de busca: " . $e;             
            }
       mysqli_close($con);
    }

//CARREGAR LISTA PÁGINA ACESSOS------------------------------------------------------------------------------------------------------------------
    function carregaListaGrupoAcesso($qtdOpcoes, $ordem){                
            $sql = "select * from grupo_acesso where ";           
            
            for ($i = 0 ; $i <= strlen((array)$qtdOpcoes) ; $i++){
                $sql = $sql . $qtdOpcoes[$i];
                    if($i != $qtdOpcoes[$i] - 1){
                        $sql = $sql . "and ";  
                    } 
            }
            
            $sql = $sql . " order by " . $ordem;
        
        try{
            $con = BD_AbrirConexao();
            $gargantula = mysqli_query($con,$sql);                
            if($carga = mysqli_fetch_array($gargantula)){
							  $ListaPaginaAcesso = [];
							  $daoPagina = new PaginaAcessoDAO();
                        while($carga = mysqli_fetch_array($gargantula)){
                              $gpAcess = new GrupoAcesso();
                              $gpAcess->setCodigo($carga["codigo"]);
                              $gpAcess->setNome($carga["nome"]);
                              $gpAcess->setTipoStatus($carga["status"]);
                              $gpAcess->setListaPaginaAcesso($daoPagina->carregarListaPaginaAcessoCod($codigo));
                              $ListaPaginaAcesso[] = $gpAcess;
                         }
                        mysqli_close($con);
                        return $ListaPaginaAcesso;
                    }else{
                        echo "Registros não encontrados!";
                    }
        }catch(Exception $e){
                echo "Erro de busca: " . $e;           
            }
mysqli_close($con); 
    }

//DELETAR----------------------------------------------------------------------------------------------------------------------------------------
    function deletarGrupoAcesso($codigo){
            try {
                $con = BD_AbrirConexao();
                $sql = "delete from grupo_acesso where codigo = " . $codigo; 
                $carga = mysqli_query ($con,$sql);
			    echo "Exclusão com sucesso";
            }catch(Exception $e){
                echo "Erro de exclusão: " . $e;
            }
	mysqli_close($con);			
    }
  }
}
?>